x-pack/plugins/observability/server/routes/copilot/route.ts

```diff
@@ -40,12 +42,91 @@ const promptRoutes: {
           throw Boom.notImplemented();
         }
 
-        return client.chatCompletion.create(prompt.messages(resources.params.body as any));
+        try {
+          return await client.chatCompletion.create(prompt.messages(resources.params.body as any));
+        } catch (error: any) {
+          if (axios.isAxiosError(error) && error.response?.status === 401) {
+            throw Boom.forbidden(error.response?.statusText);
+          }
+          throw error;
+        }
       },
     });
   })
 );
 
+const trackRoute = createObservabilityServerRoute({
+  endpoint: 'POST /internal/observability/copilot/prompts/{promptId}/track',
+  params: t.type({
+    path: t.type({
+      promptId: t.string,
+    }),
+    body: t.intersection([
+      t.type({
+        responseTime: t.number,
+        messages: t.array(
+          t.intersection([
+            t.type({
+              role: t.union([
+                t.literal(ChatCompletionRequestMessageRoleEnum.System),
+                t.literal(ChatCompletionRequestMessageRoleEnum.User),
+                t.literal(ChatCompletionRequestMessageRoleEnum.Assistant),
+              ]),
+              content: t.string,
+            }),
+            t.partial({
+              name: t.string,
+            }),
+          ])
+        ),
+        response: t.string,
+      }),
+      t.partial({
+        feedbackAction: t.union([t.literal('thumbsup'), t.literal('thumbsdown')]),
+      }),
+    ]),
+  }),
+  options: {
+    tags: [],
+  },
+  handler: async (resources): Promise<void> => {
+    const {
+      params,
+      config,
+      dependencies: {
+        pluginsSetup: { core },
+      },
+    } = resources;
+
+    if (!config.aiAssistant?.enabled) {
+      throw Boom.notImplemented;
+    }
+
+    const info = await (
+      await core.getStartServices()
+    )[0].elasticsearch.client.asInternalUser.info();
```

Comment:
Since this route doesn't have Access Tags and all authenticated users can access it, is there any concern with calling the ES client with `asInternalUser` (the system user)?

Reply:
I think the same applies here - it's a tech preview, opt-in, and the info call by itself is relatively harmless.